cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -1429,6 +1435,17 @@ def configure_apiserver(etcd_connection_string):
         elif is_state('leadership.set.keystone-cdk-addons-configured'):
             hookenv.log('Unable to find keystone endpoint. Will retry')
         remove_state('keystone.apiserver.configured')
+        # Note that we can get into a nasty state here
+        # if the user has specified webhook and they're relying on
+        # keystone auth to handle that, the api server will fail to start
+        # because we push it Webhook and no webhook config. We can't generate
+        # the config because we can't talk to the apiserver to get the ip of
+        # the service to put into the webhook template. A chicken and egg
+        # problem. To fix this, remove Webhook if keystone is related and
+        # trying to come up until we can find the service IP.
+        if 'Webhook' in auth_mode:
+            auth_mode = ','.join([i for i in auth_mode.split(',')
+                                  if i != 'Webhook'])
```

Comment:
Should this go in the `if ks and not ks_ip:` block? As it is now, this will make it impossible to use Webhook auth unless you're related to keystone, right?